lib/systems/src/TorqueControllers.cpp

```diff
@@ -153,5 +153,104 @@ void TorqueControllerLoadCellVectoring::tick(
+            writeout_.command.torqueSetpoints[RR] = brake_torque_req;
+
+            //init launch vars
+            launch_speed_target = 0;
+            time_of_launch = tick.millis;
+
+            // check speed is 0 and pedals not pressed
+            if(pedalsData.accelPercent < launch_ready_accel_threshold
+               && pedalsData.brakePercent < launch_ready_brake_threshold
+               && max_speed * RPM_TO_METERS_PER_SECOND < launch_ready_speed_threshold)
+            {
+                launch_state = LaunchStates_e::LAUNCH_READY;
+            }
+
+            break;
+        case LaunchStates_e::LAUNCH_READY:
+            // set torques and speed to 0
+            writeout_.command.speeds_rpm[FL] = 0.0;
+            writeout_.command.speeds_rpm[FR] = 0.0;
+            writeout_.command.speeds_rpm[RL] = 0.0;
+            writeout_.command.speeds_rpm[RR] = 0.0;
+
+            writeout_.command.torqueSetpoints[FL] = brake_torque_req;
+            writeout_.command.torqueSetpoints[FR] = brake_torque_req;
+            writeout_.command.torqueSetpoints[RL] = brake_torque_req;
+            writeout_.command.torqueSetpoints[RR] = brake_torque_req;
+
+            //init launch vars
+            launch_speed_target = 0;
+            time_of_launch = tick.millis;
+
+            //check speed is 0 and brake not pressed
+            if (pedalsData.brakePercent >= launch_ready_brake_threshold
+                || max_speed >= launch_ready_speed_threshold)
+            {
+                launch_state = LaunchStates_e::LAUNCH_NOT_READY;
+            } else if(pedalsData.accelPercent >= launch_go_accel_threshold){
+                launch_state = LaunchStates_e::LAUNCHING;
+            }
+
+            //check accel above launch threshold and launch
+            break;
+        case LaunchStates_e::LAUNCHING:
+            //check accel below launch threshold and brake above
+            if(pedalsData.accelPercent <= launch_ready_accel_threshold
+               || pedalsData.brakePercent >= launch_ready_brake_threshold)
+            {
+                launch_state = LaunchStates_e::LAUNCH_NOT_READY;
+            }
+
+            launch_speed_target = (int16_t)((float) (tick.millis - time_of_launch) / 1000.0 * launch_rate_target_ * 60.0 / 1.2767432544 * 11.86);
+            launch_speed_target += 1500;
+            launch_speed_target = std::min(20000, std::max(0, (int)launch_speed_target));
+
+            writeout_.command.speeds_rpm[FL] = launch_speed_target;
+            writeout_.command.speeds_rpm[FR] = launch_speed_target;
+            writeout_.command.speeds_rpm[RL] = launch_speed_target;
+            writeout_.command.speeds_rpm[RR] = launch_speed_target;
+
+            writeout_.command.torqueSetpoints[FL] = max_torque_target;
+            writeout_.command.torqueSetpoints[FR] = max_torque_target;
+            writeout_.command.torqueSetpoints[RL] = max_torque_target;
+            writeout_.command.torqueSetpoints[RR] = max_torque_target;
```

Comment:
units in `writeout_.command.torqueSetpoints` are floats with units nm. See other TCs for examples. This code currently commands 2142nm. InverterInterface handles conversion to fixed point for the inverter CAN message.